cli/src/commands/describe.rs

```diff
@@ -67,35 +77,96 @@ pub(crate) fn cmd_describe(
     args: &DescribeArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let commit = workspace_command.resolve_single_rev(&args.revision)?;
-    workspace_command.check_rewritable([commit.id()])?;
-    let description = if args.stdin {
+    let commits = workspace_command
+        .resolve_some_revsets_default_single(&args.revisions)?
+        .into_iter()
+        .collect_vec();
+    if commits.is_empty() {
+        return Err(user_error("Empty revision set"));
+    }
+    workspace_command.check_rewritable(commits.iter().ids())?;
+    let shared_description = if args.stdin {
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer).unwrap();
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        let template =
-            description_template_for_describe(ui, command.settings(), &workspace_command, &commit)?;
-        edit_description(workspace_command.repo(), &template, command.settings())?
+        None
     };
-    if description == *commit.description() && !args.reset_author {
-        writeln!(ui.status(), "Nothing changed.")?;
+
+    // TODO: Allow editing description of multiple commits in a single editor.
+    let commit_descriptions: IndexMap<_, _> = commits
```

Comment:
nit: perhaps, this can be simplified by doing `start_transaction()` earlier and rewriting commit one by one. workspace_helper can be obtained from `tx.base_workspace_helper()`.